### PR TITLE
Added 'prof' Command

### PR DESCRIPTION
**Pull request by valdeza**
This command sets equipment proficiency.

Came in handy when testing the Artisan upgrades.

### CODE REVIEW

File: src/ChannelServer/Util/GmCommands.cs [hunk 1]
```diff
@@ -2405,6 +2406,49 @@ private CommandResult HandleQuest(ChannelClient client, Creature sender, Creatur
...
+
+			if (args.Count >= 3 && !int.TryParse(args[2], out profAmount))
+			{
+				Send.ServerMessage(sender, Localization.Get("Could not parse 'amount' argument."));
+				return CommandResult.InvalidArgument;
+			}
+
+			var targetEquipment = target.Inventory.GetMainEquipment(item =>
+				((equipFlags & EquipFlag_Armor) != 0 && item.Info.Pocket == Pocket.Armor) ||
+				((equipFlags & EquipFlag_RHand) != 0 && item.Info.Pocket == (target.Inventory.WeaponSet == WeaponSet.First ? Pocket.RightHand1 : Pocket.RightHand2)) ||
```
Comment (exectails) [inline src/ChannelServer/Util/GmCommands.cs]: There's a shortcut for that pocket check, `target.Inventory.RightHandPocket` will return the correct pocket based on the weapon set. Same for the left hand of course.

**Comment by exectails** [comment]
I like the idea, but I find the bitmask a little unwieldy. I wonder if there might be a better way to go go about that, especially since someone might want to prof other items as well.

**Comment by valdeza** [comment]
You mean like proffing an item that they cannot equip? I'm not sure if the (DEV) text that appears when hovering over an item is client-side or server-side, but a clear catch-all solution here would be to pass in item entity IDs instead. 
But that would be expecting the command user to be packet sniffing for that entity ID that, as far as I know, cannot normally be shown in the client. 
Either that or be running a modded client that will display that entity item ID in the hoverbox. Trivial or not, I'm not sure; I have never made my own mods.

But if we are to stick to only equipped items, we could instead have the user name the slot to prof (e.g. "armor", "glove", "shoe", "lhand", "altrhand"; [similar to the 'go' command](https://github.com/aura-project/aura/blob/80885cca761828fe6f3f2a9c80c04fbc70b35598/src/ChannelServer/Util/GmCommands.cs#L395)).

Your thoughts?

**Comment by exectails** [comment]
I meant only equipped items, and yea, I think text based would be much easier. We have to keep in mind that numbers are hard to remember, and the concept of bitmasks isn't really obvious to everybody, especially newbies.

**Pull request merged**